cf-231A.mdx

```diff
@@ -0,0 +1,188 @@
+---
+id: cf-231A
+source: CF
+title: Team
+author: Sathvik Chundru
+---
+
+[Official Analysis (CPP)](https://codeforces.com/blog/entry/5486)
+
+# Solution (Greedy)
+
+Let's consider the following test case:
+
+```
+3
+1 1 0
+1 1 1
+1 0 0
+```
+
+The solution to finding how many of these problems that the group of friends
+will attempt is understanding exactly what the question is asking and what
+information we are given. If **two or more** friends think that they can solve
+the problems, then all three of them decide to attempt it. We know that the rows
+represent the problem and the columns represent each of the friends. A 1
+represents a problem that **one** friend feels they can do and a 0 represents a
+problem that **one** friend thinks they cannot solve. With this, we know that if
+a row contains two or more 1s, then the friends all will try that problem. We
+just need to save the total number of these types of cases and print it out at
+the end.
+
+The first solution that we will discuss is a simple as all it does is saving the
+1s and 0s into a 2-d array of size [n][3] since n represents the total number of
+problems (rows) and 3 the number of friends (columns). We can then write a for
+loop and create a temporary counter to iterate through the 2-d array n times and
+another for loop inside of it to iterate through each number on each row. If we
+find that a number is 1, we will add one to the temporary counter. At the end of
+the loop, we will check if the temporary counter is greater than or equal to
+two, and if it is, we will add one to the main counter that keeps track of
+number of problems they will attempt. Once the inner for loop resets (because
+the out for loop will have completed an iteration out of n) the temp counter
+resets back to 0, and so we can continue to reuse it. Once the outer for loop is
+done iteration, we can just print the main counter. Below are the CPP and Java
+implementations of this solution.
+
+There is another faster and simpler approach to this problem. Before you read
+the paragraph below this, think about what you can do to make this problem
+faster. What mathmatical properties were discussed in the previous solution that
+we can exploit? If you have a solid idea or are stuck, continue reading.
+
+The much simpler but more logic based solution would be to creat a for loop to
+iterate through n problems, input all 3 numbers as integers and write an if
+statement to check if the sum of the integers is greater than or equal to 2, and
+if it is, adding one to a main counter which we will then print out at the end,
+similar to the last approach. Below are the CPP and Java implementations of this
+solution.
+
+This is an example of using math to simplify and create an efficient program
+using math for greedy algorithms. Next time you ever think of an approach to
+solve a problem, consider any alternate simpler and faster algorithms you can
+create.
```

Comment:
can we make this shorter 🥶

Reply:
yeah no editorial is this long

Reply:
especially given the solution can be this short ```py
MIN_SURE = 2

print(sum(sum([int(i) for i in input().split()]) >= MIN_SURE for _ in range(int(input()))))
```

Reply:
this entire thing is unnecessary; can we shorten to one or two sentences?